core/chains/evm/gas/rollups/l1_oracle.go

```diff
@@ -47,21 +49,21 @@ func IsRollupWithL1Support(chainType chaintype.ChainType) bool {
 	return slices.Contains(supportedChainTypes, chainType)
 }
 
-func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType) (L1Oracle, error) {
+func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType, daOracle evmconfig.DAOracle) (L1Oracle, error) {
```

Comment:
as discussed, we can keep the hardcoded switch statement when daOracle does not exist for backwards-compatibility